bluesky_widgets/qt/search_input.py

```diff
@@ -16,54 +26,138 @@ def __init__(self, model, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         self.setLayout(QFormLayout())
+        # 4 Radiobuttons to quickly select default time period
+        self.all_widget = QRadioButton("All")
+        self.days_widget = QRadioButton("30 Days")
+        self.today_widget = QRadioButton("Today")
+        self.hour_widget = QRadioButton("Last Hour")
+        self.radio_button_group = QButtonGroup()
+        self.radio_button_group.addButton(self.all_widget)
+        self.radio_button_group.addButton(self.days_widget)
+        self.radio_button_group.addButton(self.today_widget)
+        self.radio_button_group.addButton(self.hour_widget)
+        default_period_layout = QGridLayout()
+        default_period_layout.addWidget(self.all_widget, 0, 0, 1, 1)
+        default_period_layout.addWidget(self.days_widget, 0, 1, 1, 1)
+        default_period_layout.addWidget(self.today_widget, 1, 0, 1, 1)
+        default_period_layout.addWidget(self.hour_widget, 1, 1, 1, 1)
+        self.layout().addRow("When:", default_period_layout)
 
         # "Since: <datetime picker>"
         self.since_widget = QDateTimeEdit()
         self.since_widget.setCalendarPopup(True)
         self.since_widget.setDisplayFormat("yyyy-MM-dd HH:mm")
         self.layout().addRow("Since:", self.since_widget)
+        self.since_widget.dateTimeChanged.connect(self.uncheck_radiobuttons)
 
         # "Until: <datetime picker>"
         self.until_widget = QDateTimeEdit()
         self.until_widget.setCalendarPopup(True)
         self.until_widget.setDisplayFormat("yyyy-MM-dd HH:mm")
         self.layout().addRow("Until:", self.until_widget)
+        self.until_widget.dateTimeChanged.connect(self.uncheck_radiobuttons)
 
+        # Refresh Button
         self.refresh_button = QPushButton("Refresh")
         self.layout().addWidget(self.refresh_button)
 
         # Initialize values.
         qdatetime = QDateTime()
-        qdatetime.setSecsSinceEpoch(self.model.since)
+        qdatetime.setSecsSinceEpoch(self.model.since.timestamp())
         self.since_widget.setDateTime(qdatetime)
         qdatetime = QDateTime()
-        qdatetime.setSecsSinceEpoch(self.model.until)
+        qdatetime.setSecsSinceEpoch(self.model.until.timestamp())
         self.until_widget.setDateTime(qdatetime)
 
         # Changes to the GUI update the model.
         self.since_widget.dateTimeChanged.connect(self.on_since_view_changed)
         self.until_widget.dateTimeChanged.connect(self.on_until_view_changed)
-        self.refresh_button.clicked.connect(self.model.events.reload)
+        # TODO: check what refresh button is doing
+        # self.refresh_button.clicked.connect(self.model.events.reload)
```

Comment:
Hm, I agree that this is subtle. After thinking about this for bit, I believe the correct "reactive" approach is to connect the button directly to `self.model.request_reload`. All the button should do is tell the model to reload.

Then, create a new method `on_reload` on this class and, in `__init__`, connect it to both `self.model.events.reload` (which fires when we have refreshed content in the same catalog) and `self.model.events.query` (which fires when we have a new catalog). Inside `on_reload`, inspect `self.model.since` and `self.model.until`. If they are timedeltas, it should compute the absolute time relative to "now" and update the since/until UIs accordingly, similar to the code in `on_since_model_changed`. In this case, we won't have `event.date` passed in, but we can consult the model directly by looking at `self.model.since`. The value of `since` will _not_ have changed, but we will reevaluate its meaning relative to "now".

Reply:
> Then, create a new method `on_reload` on this class and, in `__init__`, connect it to both `self.model.events.reload` (which fires when we have refreshed content in the same catalog) and `self.model.events.query` (which fires when we have a new catalog). 

What do you mean by 'on this class'? The SearchInput class?

Reply:
Right, exactly as in the commit you just added. 👍

Reply:
Ok, I don't get the connection right yet, as model.events.reload and query don't lead to the same event object as model.events.since or until. I get the error: AttributeError: 'Event' object has no attribute 'date'

Reply:
Oh, I understand now. See https://github.com/bluesky/bluesky-widgets/pull/13#discussion_r477485246